tests/unit/style/parsing/image.rs

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use cssparser::Parser;
+use media_queries::CSSErrorReporterTest;
+use style::parser::ParserContext;
+use style::stylesheets::Origin;
+use style::values::specified::image::*;
+use url::Url;
+
+macro_rules! assert_roundtrip_image {
```

Comment:
Can we generalize this as `assert_roundtrip_with_context` and have it take a `$fun` arg (which we use instead of `Image::parse`)